### PR TITLE
[tests-only][full-ci] implement DRY principle

### PR DESCRIPTION
**Pull request by PrajwolAmatya**
## Description
This PR implements the DRY principle on the existing code.

## Related Issue
- https://github.com/owncloud/ocis/issues/6130

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>localApiTests-apiContract-ocis</strong>](https://drone.owncloud.com/owncloud/ocis/23409/50/1) failed. Further test are cancelled...

File: tests/acceptance/features/bootstrap/Provisioning.php [hunk 1]
```diff
@@ -2420,25 +2330,6 @@ public function adminRetrievesTheInformationOfUserUsingTheProvisioningApi(
 		);
 	}
 
-	/**
-	 * @Given /^the administrator has retrieved the information of user "([^"]*)"$/
```
Comment (SwikritiT) [inline tests/acceptance/features/bootstrap/Provisioning.php:2424]: are these steps not used?

Comment (PrajwolAmatya) [reply tests/acceptance/features/bootstrap/Provisioning.php:2424]: No, these steps are not used in any feature files. There are more steps which are not in use. I have deleted some which I found while refactoring but there might be some other steps which aren't used.

Comment (SwikritiT) [reply tests/acceptance/features/bootstrap/Provisioning.php:2424]: Do you feel like these steps can be of use in the near future? I'm not entirely sure if we should remove them, there might be occasions where these steps might come in handy but it also doesn't make sense to have some unused code lying around just so "there might be occasions where these steps might come in handy".

Why were these steps added in the first place do you have any idea?

We can remove them and re-implement them whenever they are needed I guess. Or any other suggestions from other members

cc @saw-jan @SagarGi @grgprarup @amrita-shrestha

File: tests/acceptance/features/bootstrap/WebDavPropertiesContext.php [hunk 2]
```diff
@@ -1284,116 +1290,59 @@ public function theseEtagsShouldNotHaveChanged(TableNode $etagTable):void {
...
+	//      $actualEtag = $this->getCurrentEtagOfElement($path, $user);
+	//      $storedEtag = $this->getStoredEtagOfElement($path, $user, __METHOD__);
+	//      Assert::assertEquals(
+	//          $storedEtag,
+	//          $actualEtag,
+	//          __METHOD__
+	//          . " The etag of element '$path' of user '$user' was not expected to change."
+	//          . " The stored etag was '$storedEtag' but got '$actualEtag' from the response"
+	//      );
+	//  }
```
[the remark this answers is not in the document]
Comment (PrajwolAmatya) [reply tests/acceptance/features/bootstrap/WebDavPropertiesContext.php:1313]: This is supposed to be removed.

File: tests/acceptance/features/bootstrap/Provisioning.php (hunk 1 above)
Comment (SagarGi) [reply tests/acceptance/features/bootstrap/Provisioning.php:2424]: IMO we can follow YAGNI (You are not gonna need it) for now. If the step is required in future then the person can implement it again.

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 3]
```diff
@@ -2354,27 +2354,42 @@ public function userGetsAllSharesSharedWithHimFromFileOrFolderUsingTheProvisioni
...
-			$this->getSharesEndpointPath(),
+			$this->getSharesEndpointPath($endpointPath),
 			$this->getStepLineRef(),
 			[],
 			$this->ocsApiVersion
 		);
 	}
 
+	/**
+	 * @When user :user gets all shares shared by him/her using the sharing API
```
Comment (SagarGi) [inline tests/acceptance/features/bootstrap/Sharing.php:2383]:
```suggestion
	 * @When user :user gets all shares shared by (?:him|her) using the sharing API
```

can we do like this?

File: tests/acceptance/features/bootstrap/WebDavPropertiesContext.php [hunk 4]
```diff
@@ -1257,14 +1257,15 @@ public function getStoredEtagOfElement(string $path, string $user, string $messa
...
+	 * @Then /^these etags (should|should not) have changed:$/
 	 *
+	 * @param string $shouldShouldNot
 	 * @param TableNode $etagTable
 	 *
 	 * @return void
 	 * @throws Exception
 	 */
-	public function theseEtagsShouldNotHaveChanged(TableNode $etagTable):void {
+	public function theseEtagsShouldHaveChanged(string $shouldShouldNot, TableNode $etagTable):void {
```
Comment (SagarGi) [inline tests/acceptance/features/bootstrap/WebDavPropertiesContext.php:1268]:
```suggestion
	public function theseEtagsShouldOrShouldNotHaveChanged(string $shouldShouldNot, TableNode $etagTable):void {
```

File: tests/acceptance/features/bootstrap/WebDavPropertiesContext.php [hunk 5]
```diff
@@ -1285,115 +1291,36 @@ public function theseEtagsShouldNotHaveChanged(TableNode $etagTable):void {
...
 	 *
 	 * @param string $path
 	 * @param string $user
+	 * @param string $shouldShouldNot
 	 *
 	 * @return void
 	 * @throws Exception
 	 */
-	public function etagOfElementOfUserShouldNotHaveChanged(string $path, string $user):void {
+	public function etagOfElementOfUserShouldHaveChanged(string $path, string $user, string $shouldShouldNot):void {
```
Comment (SagarGi) [inline tests/acceptance/features/bootstrap/WebDavPropertiesContext.php:1303]:
```suggestion
	public function etagOfElementOfUserShouldOrShouldNotHaveChanged(string $path, string $user, string $shouldShouldNot):void {
```

File: tests/acceptance/features/bootstrap/Sharing.php (hunk 3 above)
Comment (PrajwolAmatya) [reply tests/acceptance/features/bootstrap/Sharing.php:2383]: It won't find step with `shared by him`. Probably write `(him?:him|her)`?

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 6]
```diff
@@ -570,45 +570,15 @@ public function theUserLooksUpTheSingleSpaceUsingTheGraphApiByUsingItsId(string
...
-				$body
-			)
-		);
-
-		$this->setSpaceCreator($spaceName, $user);
-	}
-
 	/**
 	 * @When /^user "([^"]*)" creates a space "([^"]*)" of type "([^"]*)" with quota "([^"]*)" using the Graph API$/
 	 * @When /^user "([^"]*)" tries to create a space "([^"]*)" of type "([^"]*)" with quota "([^"]*)" using the Graph API$/
```
Comment (grgprarup) [inline tests/acceptance/features/bootstrap/SpacesContext.php:575]:
```suggestion
	 * @When /^user "([^"]*)" (?:creates|tries to create) a space "([^"]*)" of type "([^"]*)" with quota "([^"]*)" using the Graph API$/
```
Can we change here too?

File: tests/acceptance/features/bootstrap/ArchiverContext.php [hunk 7]
```diff
@@ -151,6 +140,28 @@ public function userDownloadsTheArchiveOfItemOfUser(
...
+		string $downloader,
+		string $resource,
+		string $addressType,
+		?string $owner = null,
+		?array $headers = null
+	): void {
+		$owner = $owner ??  $downloader;
 		$downloader = $this->featureContext->getActualUsername($downloader);
 		$queryString = $this->getArchiverQueryString($owner, $resource, $addressType);
 		$this->featureContext->setResponse(
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/ArchiverContext.php:167]:
```suggestion
		return
```

File: tests/acceptance/features/bootstrap/ArchiverContext.php [hunk 8]
```diff
@@ -159,6 +170,7 @@ public function userDownloadsTheArchiveOfItemOfUser(
 				'',
 				$downloader,
 				$this->featureContext->getPasswordForUser($downloader),
+				$headers
 			)
 		);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/ArchiverContext.php:175]:
```suggestion
```

File: tests/acceptance/features/bootstrap/ArchiverContext.php [hunk 9]
```diff
@@ -118,18 +118,7 @@ public function userDownloadsTheArchive(
...
-		$this->featureContext->setResponse(
-			HttpRequestHelper::get(
-				$this->featureContext->getBaseUrl() . '/archiver?' . $queryString,
-				'',
-				$user,
-				$this->featureContext->getPasswordForUser($user),
-				$headers
-			)
-		);
+		$this->downloadArchive($user, $resource, $addressType, null, $headers);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/ArchiverContext.php:121]:
```suggestion
		$response = $this->downloadArchive($user, $resource, $addressType, null, $headers);
		$this->featureContext->setResponse($response);
```

File: tests/acceptance/features/bootstrap/ArchiverContext.php [hunk 10]
```diff
@@ -151,6 +140,28 @@ public function userDownloadsTheArchiveOfItemOfUser(
 		string $owner,
 		string $addressType
 	): void {
+		$this->downloadArchive($downloader, $resource, $addressType, $owner);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/ArchiverContext.php:143]:
```suggestion
		$response = $this->downloadArchive($downloader, $resource, $addressType, $owner);
		$this->featureContext->setResponse($response);
```

**Review by saw-jan — CHANGES_REQUESTED** [review]
This is a helper function and we should not save the response in any helpers - saving might lead to unexpected behaviors. We should only save response in the When steps.

File: tests/acceptance/features/bootstrap/FilesVersionsContext.php [hunk 11]
```diff
@@ -81,9 +68,27 @@ public function userTriesToGetFileVersions(string $user, string $file, string $f
...
+	/***
+	 * @param string $user
+	 * @param string $file
+	 * @param string|null $fileOwner
+	 *
+	 * @return void
+	 * @throws JsonException
+	 * @throws GuzzleException
+	 */
+	public function getFileVersions(
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/FilesVersionsContext.php:83]: Same with this function also

File: tests/acceptance/features/bootstrap/FilesVersionsContext.php [hunk 12]
```diff
@@ -81,9 +68,27 @@ public function userTriesToGetFileVersions(string $user, string $file, string $f
 	 * @throws Exception
 	 */
 	public function userGetsFileVersions(string $user, string $file):void {
+		$this->getFileVersions($user, $file);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/FilesVersionsContext.php:71]:
```suggestion
		$response = $this->getFileVersions($user, $file);
		$this->featureContext->setResponse($response);
```

**Review by saw-jan — COMMENTED** [review]
Same with this function

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 13]
```diff
@@ -2354,34 +2354,49 @@ public function userGetsAllSharesSharedWithHimFromFileOrFolderUsingTheProvisioni
 		);
 	}
 
-	/**
-	 * @When user :user gets all shares shared by him/her using the sharing API
-	 *
+	/***
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/Sharing.php:2357]:
```suggestion
	/**
```

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 14]
```diff
@@ -2354,34 +2354,49 @@ public function userGetsAllSharesSharedWithHimFromFileOrFolderUsingTheProvisioni
...
+	 * @throws JsonException
+	 * @throws GuzzleException
 	 */
-	public function userGetsAllSharesSharedByHimUsingTheSharingApi(string $user):void {
+	public function getAllShares(
+		string $user,
+		?string $endpointPath = null
+	): void {
 		$user = $this->getActualUsername($user);
 		$this->response = OcsApiHelper::sendRequest(
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/Sharing.php:2370]:
```suggestion
		return OcsApiHelper::sendRequest(
```

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 15]
```diff
@@ -2354,34 +2354,49 @@ public function userGetsAllSharesSharedWithHimFromFileOrFolderUsingTheProvisioni
...
 	/**
-	 * @When the administrator gets all shares shared by him/her using the sharing API
+	 * @When /^user "([^"]*)" gets all shares shared by (?:him|her) using the sharing API$/
+	 *
+	 * @param string $user
+	 *
+	 * @return void
+	 */
+	public function userGetsAllSharesSharedByHimUsingTheSharingApi(string $user):void {
+		$this->getAllshares($user);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/Sharing.php:2390]:
```suggestion
		$this->setResponse($this->getAllshares($user));
```

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 16]
```diff
@@ -2354,34 +2354,49 @@ public function userGetsAllSharesSharedWithHimFromFileOrFolderUsingTheProvisioni
...
+	}
+
+	/**
+	 * @When /^the administrator gets all shares shared by (?:him|her) using the sharing API$/
 	 *
 	 * @return void
 	 */
 	public function theAdministratorGetsAllSharesSharedByHimUsingTheSharingApi():void {
-		$this->userGetsAllSharesSharedByHimUsingTheSharingApi($this->getAdminUsername());
+		$this->getAllShares($this->getAdminUsername());
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/Sharing.php:2399]:
```suggestion
		$this->setResponse($this->getAllShares($this->getAdminUsername());
```

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 17]
```diff
@@ -2402,16 +2416,7 @@ public function userGetsFilteredSharesSharedByHimUsingTheSharingApi(string $user
...
-			$this->getBaseUrl(),
-			$user,
-			$this->getPasswordForUser($user),
-			"GET",
-			$this->getSharesEndpointPath("?share_types=" . $rawShareTypes),
-			$this->getStepLineRef(),
-			[],
-			$this->ocsApiVersion
-		);
+		$this->getAllShares($user, "?share_types=" . $rawShareTypes);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/Sharing.php:2419]:
```suggestion
		$response = $this->getAllShares($user, "?share_types=" . $rawShareTypes);
		$this->setResponse($response);
```

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 18]
```diff
@@ -2423,17 +2428,7 @@ public function userGetsFilteredSharesSharedByHimUsingTheSharingApi(string $user
...
-			$this->getBaseUrl(),
-			$user,
-			$this->getPasswordForUser($user),
-			"GET",
-			$this->getSharesEndpointPath("?path=$path"),
-			$this->getStepLineRef(),
-			[],
-			$this->ocsApiVersion
-		);
+		$this->getAllShares($user, "?path=$path");
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/Sharing.php:2431]:
```suggestion
		$response = $this->getAllShares($user, "?path=$path");
		$this->setResponse($response);
```

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 19]
```diff
@@ -2448,17 +2443,7 @@ public function userGetsAllTheSharesWithResharesFromTheFileUsingTheSharingApi(
...
-			$this->getBaseUrl(),
-			$userActual,
-			$this->getPasswordForUser($user),
-			"GET",
-			$this->getSharesEndpointPath("?reshares=true&path=$path"),
-			$this->getStepLineRef(),
-			[],
-			$this->ocsApiVersion
-		);
+		$this->getAllShares($user, "?reshares=true&path=$path");
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/Sharing.php:2446]:
```suggestion
		$response = $this->getAllShares($user, "?reshares=true&path=$path");
		$this->setResponse($response);
```

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 20]
```diff
@@ -2470,17 +2455,7 @@ public function userGetsAllTheSharesWithResharesFromTheFileUsingTheSharingApi(
...
-			$this->getBaseUrl(),
-			$user,
-			$this->getPasswordForUser($user),
-			"GET",
-			$this->getSharesEndpointPath("?path=$path&subfiles=true"),
-			$this->getStepLineRef(),
-			[],
-			$this->ocsApiVersion
-		);
+		$this->getAllShares($user, "?path=$path&subfiles=true");
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/Sharing.php:2458]:
```suggestion
		$response = $this->getAllShares($user, "?path=$path&subfiles=true");
		$this->setResponse($response);
```

**Review by saw-jan — COMMENTED** [review]
Save response only from When steps

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 21]
```diff
@@ -1275,23 +1215,20 @@ public function theUserUploadsAFileToAnotherOwnerSpace(
 		$this->featureContext->uploadFileWithContent($user, $content, $destination);
 	}
 
-	/**
-	 * @When /^user "([^"]*)" (?:changes|tries to change) the name of the "([^"]*)" space to "([^"]*)"$/
-	 * @When /^user "([^"]*)" (?:changes|tries to change) the name of the "([^"]*)" space to "([^"]*)" owned by user "([^"]*)"$/
-	 *
+	/***
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:1218]:
```suggestion
	/**
```

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 22]
```diff
@@ -1302,7 +1239,6 @@ public function updateSpaceName(
 			$spaceId = $space["id"];
 		}
 
-		$bodyData = ["Name" => $newName];
 		$body = json_encode($bodyData, JSON_THROW_ON_ERROR);
 
 		$this->featureContext->setResponse(
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:1244]:
```suggestion
		return
```

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 23]
```diff
@@ -1316,6 +1252,29 @@ public function updateSpaceName(
...
+	 * @throws Exception
+	 */
+	public function updateSpaceName(
+		string $user,
+		string $spaceName,
+		string $newName,
+		string $owner = ''
+	): void {
+		$bodyData = ["Name" => $newName];
+		$this->updateSpace($user, $spaceName, $bodyData, $owner);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:1275]:
```suggestion
		$response = $this->updateSpace($user, $spaceName, $bodyData, $owner);
		$this->featureContext->setResponse($response);
```

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 24]
```diff
@@ -1335,26 +1294,8 @@ public function updateSpaceDescription(
...
-		$this->featureContext->setResponse(
-			GraphHelper::updateSpace(
-				$this->featureContext->getBaseUrl(),
-				$user,
-				$this->featureContext->getPasswordForUser($user),
-				$body,
-				$spaceId
-			)
-		);
+		$this->updateSpace($user, $spaceName, $bodyData, $owner);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:1298]:
```suggestion
		$response = $this->updateSpace($user, $spaceName, $bodyData, $owner);
		$this->featureContext->setResponse($response);
```

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 25]
```diff
@@ -1399,26 +1340,8 @@ public function updateSpaceQuota(
...
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:1344]:
```suggestion
		$response = $this->updateSpace($user, $spaceName, $bodyData, $owner);
		$this->featureContext->setResponse($response);
```

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 26]
```diff
@@ -1561,6 +1472,23 @@ public function theUserHasCreatedASpaceByDefaultUsingTheGraphApi(
...
+	 * @throws GuzzleException
+	 * @throws JsonException
+	 */
+	public function createSpace(
+		string $user,
+		string $spaceName,
+		array $space
+	): void {
 		$body = json_encode($space, JSON_THROW_ON_ERROR);
 		$this->featureContext->setResponse(
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:1493]:
```suggestion
		return
```

**Review by saw-jan — COMMENTED** [review]
no setResponse in helper functions

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 27]
```diff
@@ -1530,19 +1453,7 @@ public function userHasCreatedSpace(
...
-				$user,
-				$this->featureContext->getPasswordForUser($user),
-				$body
-			)
-		);
-		$this->featureContext->theHTTPStatusCodeShouldBe(
-			201,
-			"Expected response status code should be 201 (Created)"
-		);
+		$this->createSpace($user, $spaceName, $space);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:1456]:
```suggestion
		$response = $this->createSpace($user, $spaceName, $space);
		// Assert the 201 status code
```

File: tests/acceptance/features/bootstrap/TrashbinContext.php [hunk 28]
```diff
@@ -307,50 +307,36 @@ public function userGetsFilesInTheTrashbinWithDepthUsingTheWebdavApi(string $use
...
 
 		foreach ($table->getHash() as $row) {
 			$path = trim($row['name'], "/");
-			$found = false;
+			if ($shouldShouldNot !== 'not') {
+				$found = false;
+			} else {
+				$found = true;
+			}
+
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/TrashbinContext.php:331]: should we do this? :thinking:

File: tests/acceptance/features/bootstrap/WebDavPropertiesContext.php [hunk 29]
```diff
@@ -1275,7 +1276,12 @@ public function theseEtagsShouldNotHaveChanged(TableNode $etagTable):void {
...
 			$actualEtag = $this->getCurrentEtagOfElement($path, $user);
 			$storedEtag = $this->getStoredEtagOfElement($path, $user, __METHOD__);
-			if ($actualEtag !== $storedEtag) {
+			if ($shouldShouldNot === 'should' && $actualEtag === $storedEtag) {
+				$changedEtagCount = $changedEtagCount + 1;
+				$changedEtagMessage
+					.= "\nThe etag '$storedEtag' of element '$path' of user '$user' did not change.";
+			}
+			if ($shouldShouldNot === 'should not' && $actualEtag !== $storedEtag) {
 				$changedEtagCount = $changedEtagCount + 1;
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/WebDavPropertiesContext.php:1285]:
```suggestion
		        $changedEtagCount = $changedEtagCount + 1;
			if ($shouldShouldNot === 'should' && $actualEtag === $storedEtag) {
				$changedEtagMessage
					.= "\nThe etag '$storedEtag' of element '$path' of user '$user' did not change.";
			}
			if ($shouldShouldNot === 'should not' && $actualEtag !== $storedEtag) {
```

File: tests/acceptance/features/bootstrap/WebDavPropertiesContext.php [hunk 30]
```diff
@@ -1285,115 +1291,36 @@ public function theseEtagsShouldNotHaveChanged(TableNode $etagTable):void {
 	}
 
 	/**
-	 * @Then the etag of element :path of user :user should not have changed
+	 * @Then /^the etag of element "([^"]*)" of user "([^"]*)" should (|not)\s?have changed$/
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/WebDavPropertiesContext.php:1294]:
```suggestion
	 * @Then /^the etag of element "([^"]*)" of user "([^"]*)" should (not )?have changed$/
```

File: tests/acceptance/features/bootstrap/TrashbinContext.php [hunk 31]
```diff
@@ -307,50 +307,36 @@ public function userGetsFilesInTheTrashbinWithDepthUsingTheWebdavApi(string $use
 	}
 
 	/**
-	 * @Then the trashbin DAV response should not contain these nodes
+	 * @Then /^the trashbin DAV response should (not|)\s?contain these nodes$/
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/TrashbinContext.php:310]: we should prefer readability also
```suggestion
	 * @Then /^the trashbin DAV response (should|should not) contain these nodes$/
```

File: tests/acceptance/features/bootstrap/WebDavPropertiesContext.php (hunk 29 above)
Comment (PrajwolAmatya) [reply tests/acceptance/features/bootstrap/WebDavPropertiesContext.php:1285]: Moving `$changedEtagCount = $changedEtagCount + 1;` outside of the if block will fail while asserting.

Comment (saw-jan) [reply tests/acceptance/features/bootstrap/WebDavPropertiesContext.php:1285]: Ahh! double if. I thought nested if

Comment (PrajwolAmatya) [reply tests/acceptance/features/bootstrap/WebDavPropertiesContext.php:1285]: Should we keep this way or revert it to original? @saw-jan

**Review by SagarGi — COMMENTED** [review]
seems good refactor that the response is not set in the helper function but the step implementation. Overall LGTM :+1:

File: tests/acceptance/features/bootstrap/WebDavPropertiesContext.php [hunk 32]
```diff
@@ -666,19 +666,7 @@ public function valueOfItemOfPathShouldBe(string $user, string $xpath, string $p
...
-				__METHOD__
-			);
-		}
-		$xmlPart = $resXml->xpath($xpath);
-		Assert::assertTrue(
-			isset($xmlPart[0]),
-			"Cannot find item with xpath \"$xpath\""
-		);
-		$value = $xmlPart[0]->__toString();
+		$value = $this->getResponseXml($xpath);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/WebDavPropertiesContext.php:669]:
```suggestion
                $resXml = $this->featureContext->getResponseXmlObject();
		if ($resXml === null) {
			$resXml = HttpRequestHelper::getResponseXml(
				$this->featureContext->getResponse(),
				__METHOD__
			);
		}
		$value = $this->getXmlItemByXpath($resXml, $xpath);
```

File: tests/acceptance/features/bootstrap/WebDavPropertiesContext.php [hunk 33]
```diff
@@ -744,6 +720,30 @@ public function assertValueOfItemInResponseAboutUserIsEitherOr(string $xpath, ?s
...
+				__METHOD__
+			);
+		}
+		$xmlPart = $resXml->xpath($xpath);
+		Assert::assertTrue(
+			isset($xmlPart[0]),
+			"Cannot find item with xpath \"$xpath\""
+		);
+		return $xmlPart[0]->__toString();
+	}
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/WebDavPropertiesContext.php:745]:
```suggestion
	 * @param <type> $xmlResponse
	 * @param string $xpath
	 *
	 * @return string
	 * @throws Exception
	 */
	public function getXmlItemByXpath(
	        <type> $xmlResponse,
		string $xpath
	): string {
		$xmlPart = $xmlResponse->xpath($xpath);
		Assert::assertTrue(
			isset($xmlPart[0]),
			"Cannot find item with xpath \"$xpath\""
		);
		return $xmlPart[0]->__toString();
	}
```

**Review by saw-jan — COMMENTED** [review]
Let's reduce the external dependency wherever posssible

**Review by saw-jan — APPROVED** [review]
LGTM

**Pull request merged**